### DISCUSSION TITLE
Gradle version catalogs lint reports for unused dependencies workarounds

### DISCUSSION THREAD

**Discussion by JackEblan** [android/nowinandroid#1226] [Q&A] [answered]
Do you have a way to make the gradle version catalogs dependencies as being used and not greyed out? 

For example, when we declare a dependency into the convention plugins but gradle version catalogs doesn't treat this as being declared and reports as being unused.

I hope someone can provide a way for this so my lint will no longer complain, and not by supressing it😂

**Comment by SimonMarquis** [comment, marked as answer]
This is an Android Studio "feature" to detect used references.
Unfortunately, when using untyped references like what is done in convention plugins:
```kotlin
extensions.getByType<VersionCatalogsExtension>().named("libs").findLibrary(alias)
```
Android Studio can't magically make the connection between this alias and the `libs.versions.toml` file.

**Reply by JackEblan** [reply]
Sad to know it😔
